harness/tests/determined/common/experimental/test_resource_pool.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+import responses
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.common.experimental import resource_pool
+from tests.fixtures import api_responses
+
+_MASTER = "http://localhost:8080"
+
+
+@pytest.fixture
+def standard_session() -> api.Session:
+    return api.Session(master=_MASTER, user=None, auth=None, cert=None)
+
+
+@pytest.fixture
+def single_item_resource_pools() -> bindings.v1GetResourcePoolsResponse:
+    sample_resource_pools = api_responses.sample_get_resource_pool().resourcePools
+    single_item_pagination = bindings.v1Pagination(endIndex=1, startIndex=0, total=1)
+    return bindings.v1GetResourcePoolsResponse(
+        resourcePools=sample_resource_pools, pagination=single_item_pagination
+    )
+
+
+@responses.activate
+def test_bind_rp_to_workspaces_errors_on_invalid_rps(
```

Comment:
I'm going to have to think about this one a little longer (got another meeting right now). I'm not sure what behavioral contracts we want to guarantee with this test.

Reply:
First (and the easiest thing to write about and address), if the test is specifically testing that an `APIException` is raised, it should be done via `with pytest.raises...`.

Reply:
Different behaviors we might want to guarantee (from behavior we have control over at this layer):
1. The `ResourcePool` constructor will throw an exception rather than create an object if an invalid name is given
2. `bind`, `unbind`, etc. will throw an exception if the bindings call constructed from them raises an exception.

I'm not sure we can do much else.

For (1), we don't currently do this now, but if you'd like to add this functionality I'd support it. And if you do add it, I think it deserves a unit test.

For (2), I don't think "don't catch exceptions passed by the bindings" is behavior worthy of a unit test. But I have a vague sense I'm not seeing the entire picture here, so if you see this different, let's talk about it in text/huddle.

Reply:
hmm right now, there isn't an easy, RBAC-ful way to check if the resource pool is valid on startup, so I'm inclined to go with option 2. If that's the case, I'm ok removing this test

Reply:
I'm in -- let's kill it.